src/generators/java/renderers/ClassRenderer.ts

```diff
@@ -80,13 +80,35 @@ ${this.indent(this.renderBlock(content, 2))}
   runSetterPreset(property: ConstrainedObjectPropertyModel): Promise<string> {
     return this.runPreset('setter', { property });
   }
+
+  getConstValue(property: ConstrainedObjectPropertyModel): string | undefined {
+    if (!property.hasConstValue()) {
+      return undefined;
+    }
+
+    const constrainedEnumValueModel = property.getConstrainedEnumValueModel();
+
+    if (constrainedEnumValueModel) {
+      return `private final ${property.property.type} ${property.propertyName} = ${property.property.type}.${constrainedEnumValueModel.key};`;
+    } else if (property.isConstrainedStringModel()) {
+      return `private final ${property.property.type} ${
+        property.propertyName
+      } = "${property.getConstValue()}";`;
+    }
+  }
 }
 
 export const JAVA_DEFAULT_CLASS_PRESET: ClassPresetType<JavaOptions> = {
   self({ renderer }) {
     return renderer.defaultSelf();
   },
-  property({ property }) {
+  property({ renderer, property }) {
+    const constValue = renderer.getConstValue(property);
```

Comment:
What I refer to in my review is that `property.const` would be much better and follow how we currently handle the meta models.